lib/qesapdeployment.pm

```diff
@@ -745,4 +749,135 @@ sub qesap_get_az_resource_group {
     return $result;
 }
 
+=head3 qesap_get_az_address_range
+
+Query the existing maintenance deployments for azure
+and determine a name with a free adress range number
+for the current deployment
+
+=cut
+
+sub qesap_get_az_address_range {
+    my $worker = get_var("WORKER_INSTANCE");
+
+    my $offset = ($worker - 1) * 8;
+
+    my $ip2 = int($offset / 256);
+    my $ip3 = $offset % 256;
+
+    my $vnet_address_range = sprintf("10.%d.%d.0/21", $ip2, $ip3);
+    my $subnet_address_range = sprintf("10.%d.%d.0/24", $ip2, $ip3);
+
+    return (
+        vnet_address_range => $vnet_address_range,
+        subnet_address_range => $subnet_address_range,
+    );
+}
+
+=head3 qesap_az_vnet_peering
+
+    Performs peering between the cluster and IBS mirror on Azure.
+
+=over 3
+
+=item B<SOURCE_GROUP> - resource group of source
+
+=item B<SOURCE_VNET> - vnet of source
+
+=item B<TARGET_GROUP> - resource group of target
+
+=item B<TARGET_VNET> - vnet of target
+
+=item B<VERBOSE> - wether output will be verbose or not
+
+=back
+=cut
+
+sub qesap_az_vnet_peering {
+    my (%args) = @_;
+    croak 'Missing mandatory source_group argument' unless $args{source_group};
+    croak 'Missing mandatory source_vnet argument' unless $args{source_vnet};
+    croak 'Missing mandatory target_group argument' unless $args{target_group};
+    croak 'Missing mandatory target_vnet argument' unless $args{target_vnet};
+    my $az_output = "none";
+
+    if ($args{verbose}) {
+        $az_output = "table";
+    }
+
+    my $az_installed = script_run("which az");
+    if ($az_installed != 0) {
+        record_info("AZURE MISSING", "Azure command \"az\" doesn't seem installed\n");
+        return 1;
+    }
+
+    my $source_vnet_id = script_output("az network vnet show --resource-group $args{source_group} --name $args{source_vnet} --query id --output tsv");
+    record_info("[M] source vnet ID: $source_vnet_id\n") if $args{verbose};
+
+    my $target_vnet_id = script_output("az network vnet show --resource-group $args{target_group} --name $args{target_vnet} --query id --output tsv");
+    record_info("[M] target vnet ID: $target_vnet_id\n") if $args{verbose};
+
+    my $peering_name = "$args{source_vnet}-$args{target_vnet}";
+
+    my $result = script_run("az network vnet peering create --name $peering_name --resource-group $args{source_group} --vnet-name $args{source_vnet} --remote-vnet $target_vnet_id --allow-vnet-access --output $az_output");
+    if ($result == 0) {
```

Comment:
I think you can remove this if/else and just convert `script_run` to `assert_script_run`